### PR TITLE
124 Lock seed transfer

### LINKED ISSUE #124: Bug: project owners can send seed token to Seed after it is deployed.
When we are deploying the ```Seed``` contract, we are calculating the amount of ```seedToken``` required based on ```hardCap``` and transferring the ```seedToken``` to deployed Seed Contract. 

Let say when Seed contract have ```x``` amount of ```seedToken``` after it is deployed by ```SeedFactory```. Now the project owner can send more ```seedToken``` to Seed Contract. This will change the amount of ```seedToken``` with ```Seed``` Contract to ```y```. This will cause some error while calculating the remaining amount of ```seedToken```.
Even with this problem, the buyers, won't be able to buy() more than ```hardCap```.
The main problem is with the remaining seed amount calculation.

Probable solution:- create a state variable of ```Seed``` for initially allocated ```eedToken```.

### PR DESCRIPTION
**Pull request by codebuster22**
Closes #124 
Added a contract state to keep track of remaining seed tokens with Seed Contract.
Also tried removing some magic values from Seed.js

### CODE REVIEW

**Review by kinspotting — CHANGES_REQUESTED** [review]
needs changes

File: contracts/seed/Seed.sol [hunk 1]
```diff
@@ -150,13 +151,15 @@ contract Seed {
...
         minimumReached  = false;
+        remainingSeeds  = IERC20(_tokens[0]).balanceOf(address(this));
     }
 
     /**
       * @dev                     Buy seed tokens.
       * @param _seedAmount       The amount of seed tokens to buy.
     */
     function buy(uint256 _seedAmount) public protected checked {
+        remainingSeeds -=_seedAmount;
```
[the remark this answers is not in the document]
Comment (codebuster22) [reply contracts/seed/Seed.sol:162]: forgot, had previously worked with 0.8.0, will keep in mind to use SafeMath it.

File: contracts/seed/SeedFactory.sol [hunk 2]
```diff
@@ -114,6 +114,15 @@ contract SeedFactory is CloneFactory {
 
         Seed(_newSeed).updateMetadata(_metadata);
 
+        {
+            uint reqSeedAmount = (_softAndHardCap[1].div(_price)).mul(10**18);
```
[the remark this answers is not in the document]
Comment (codebuster22) [reply contracts/seed/SeedFactory.sol:119]: done.

**Review by kinspotting — APPROVED** [review]
lgtm

**Pull request merged**